Source/Core/DolphinWX/Frame.cpp

```diff
@@ -1465,3 +1466,65 @@ void CFrame::ParseHotkeys()
 	if (IsHotkey(HK_UNDO_SAVE_STATE))
 		State::UndoSaveState();
 }
+
+void CFrame::HandleFrameSkipHotkeys()
+{
+	static const int MAX_FRAME_SKIP_DELAY = 60;
+	static int frameStep = 0;
+	static int frameStepDelay = 30;
+	static int skipFrameStepDelay = 1;
+	static int skipFrameStepDelayCount = 0;
+	static bool skipFrameStep = false;
+
+	if (IsHotkey(HK_FRAME_ADVANCE_DECREASE_SPEED))
+	{
+		++skipFrameStepDelay;
+		if (skipFrameStepDelay > MAX_FRAME_SKIP_DELAY)
+			skipFrameStepDelay = MAX_FRAME_SKIP_DELAY;
+	}
+	else if (IsHotkey(HK_FRAME_ADVANCE_INCREASE_SPEED))
+	{
+		--skipFrameStepDelay;
+		if (skipFrameStepDelay < 0)
+			skipFrameStepDelay = 0;
+	}
+	else if (IsHotkey(HK_FRAME_ADVANCE_RESET_SPEED))
+	{
+		skipFrameStepDelay = 1;
+	}
+	else if (IsHotkey(HK_FRAME_ADVANCE, true))
+	{
+		if (skipFrameStepDelayCount < skipFrameStepDelay && skipFrameStep)
+			++skipFrameStepDelayCount;
+
+		if ((frameStep == 0 || frameStep == frameStepDelay) && !skipFrameStep)
+		{
+			wxCommandEvent evt;
+			evt.SetId(IDM_FRAMESTEP);
+			CFrame::OnFrameStep(evt);
+			if (skipFrameStepDelay > 0)
+				skipFrameStep = true;
+		}
+
+		if (frameStep < frameStepDelay)
+		{
+			++frameStep;
+			if (skipFrameStep)
+				skipFrameStep = false;
+		}
+
+		if (frameStep == frameStepDelay && skipFrameStep && skipFrameStepDelayCount == skipFrameStepDelay)
```

Comment:
Should use `skipFrameStepDelayCount >= skipFrameStepDelay` instead of `==` in case `skipFrameStepDelay` is modified while the frame advance key is being held.